code/modules/martial_arts/cqc.dm

```diff
@@ -12,7 +12,7 @@
 	name = "Close Quarters Cooking"
 
 /datum/martial_art/cqc/under_siege/teach(mob/living/carbon/human/H, make_temporary)
-	RegisterSignal(H, COMSIG_AREA_ENTERED, PROC_REF(kitchen_check))
+	RegisterSignal(H, COMSIG_AREA_ENTERED, PROC_REF(kitchen_check), override = TRUE)
```

Comment:
I'm more concerned about `teach()` being ran more than once on the same person. Which is the only way the signal override runtime would trigger. I wonder if this is a problem with all martial arts.